control-plane-operator/controllers/hostedcontrolplane/kcm/params.go

```diff
@@ -103,10 +102,6 @@ func NewKubeControllerManagerParams(ctx context.Context, hcp *hyperv1.HostedCont
 	params.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
 
 	switch hcp.Spec.Platform.Type {
-	case hyperv1.AWSPlatform:
-		params.CloudProvider = aws.Provider
```

Comment:
shouldn't we be setting `params.CloudProvider` to `external`?

Reply:
my understanding is that is already getting set on the kubelet by the mco when we generate an ignition payload

Reply:
right, but according to the [docs](https://cloud-provider-aws.sigs.k8s.io/getting_started/) we should set it on the kube-controller-manager as well as on the kubelet.

Reply:
ahh, ic what you mean, thx
If the current run passes, I'd rather merge it so we can get CI unstuck and I can update in a followup.

Reply:
I think this should be fine as it's, i.e. just omitting the cloud provider flag.
Let's believe these docs for now https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager and be consistent with what we've been doing for kubevirt.

```
kube-apiserver and kube-controller-manager MUST NOT specify the --cloud-provider flag. This ensures that it does not run any cloud specific loops that would be run by cloud controller manager. In the future, this flag will be deprecated and removed.
kubelet must run with --cloud-provider=external. This is to ensure that the kubelet is aware that it must be initialized by the cloud controller manager before it is scheduled any work.
```